bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java

```diff
@@ -252,54 +269,132 @@ private void scheduleInitialAccess(BoschHttpClient httpClient) {
         }
     }
 
+    /**
+     * Check the bridge access by sending an HTTP request.
+     * Does not throw any exception in case the request fails.
+     */
+    public boolean checkBridgeAccess() {
+        @Nullable
+        BoschHttpClient httpClient = this.httpClient;
+
+        if (httpClient == null) {
+            return false;
+        }
+
+        try {
+            logger.debug("Sending http request to BoschSHC to check access: {}", httpClient);
+            String url = httpClient.getBoschSmartHomeUrl("devices");
+            ContentResponse contentResponse = httpClient.createRequest(url, GET).send();
+
+            // check HTTP status code
+            if (!HttpStatus.getCode(contentResponse.getStatus()).isSuccess()) {
+                logger.debug("Access check failed with status code: {}", contentResponse.getStatus());
+                return false;
+            }
+
+            // Access OK
+            return true;
+        } catch (TimeoutException | ExecutionException | InterruptedException e) {
+            logger.warn("Access check failed because of {}!", e.getMessage());
```

Comment:
Shouldn't we add the exception as an argument to the logger call so we can see the stack trace in the log?

Reply:
I followed the [code guideline/logging](https://www.openhab.org/docs/developer/guidelines.html#f-logging) and a catched exception at this place not a bug in the binding.
We can even discuss if the warning level should be changed to ``info``, because this line is a left over from the original ``getRooms()`` code.